pilot/pkg/config/kube/gateway/deploymentcontroller.go

```diff
@@ -331,6 +331,14 @@ func NewDeploymentController(
 		}
 	}))
 
+	dc.env.Watcher.AddMeshHandler(func() {
```

Comment:
Does it need to be every meshconfig change? Redeploys aren't zero risk; can we only re-reconcile if certain fields are different from what we passed in?